src/runner/PlutoRunner.jl

```diff
@@ -2282,8 +2333,31 @@ function with_io_to_logs(f::Function; enabled::Bool=true, loglevel::Logging.LogL
     # `String`. We need to use an asynchronous task to continously tranfer bytes from the
     # pipe to `output` in order to avoid the buffer filling up and stalling write() calls in
     # user code.
+    execution_done = Ref(false)
     output = IOBuffer()
-    buffer_redirect_task = @async write(output, pipe)
+    function send_output()
+        output_str = String(take!(output))
+        if !isempty(output_str)
+            Logging.@logmsg loglevel output_str
+        end
+    end
+    @async begin
+        pipe_reader = Base.pipe_reader(pipe)
+        try
+            while !eof(pipe_reader)
```

Comment:
I expected this short `while` loop to consume 100% CPU, but it doesn't. Nice! Do you know why?

Reply:
Is `readavailable` blocking?

Reply:
From my understanding this is blocking until at least one byte is available (with `wait_readnb(this,1)`):
```julia
function readavailable(this::LibuvStream)
    wait_readnb(this, 1) # unlike the other `read` family of functions, this one doesn't guarantee error reporting
    iolock_begin()
    buf = this.buffer
    @assert buf.seekable == false
    bytes = take!(buf)
    iolock_end()
    return bytes
end
```